mir/src/passes/unrolling/unrolling_third_pass.rs

```diff
@@ -0,0 +1,468 @@
+                | MirValue::PeriodicColumn(_)
+                | MirValue::PublicInput(_)
+                | MirValue::PublicInputTable(_)
+                | MirValue::RandomValue(_)
+                | MirValue::BusAccess(_)
+                | MirValue::Null
+                | MirValue::Unconstrained => {},
+            }
+        }
+
+        Ok(updated_value)
+    }
+
+    fn visit_add_bis(
+        &mut self,
+        _graph: &mut Graph,
+        add: Link<Op>,
+    ) -> Result<Option<Link<Op>>, CompileError> {
+        // safe to unwrap because we just dispatched on it
+        let add_ref = add.as_add().unwrap();
+        let lhs = add_ref.lhs.clone();
+        let rhs = add_ref.rhs.clone();
+
+        unroll_binary_op(lhs, rhs, add.clone(), add_ref.span())
+    }
+    fn visit_sub_bis(
+        &mut self,
+        _graph: &mut Graph,
+        sub: Link<Op>,
+    ) -> Result<Option<Link<Op>>, CompileError> {
+        // safe to unwrap because we just dispatched on it
+        let sub_ref = sub.as_sub().unwrap();
+        let lhs = sub_ref.lhs.clone();
+        let rhs = sub_ref.rhs.clone();
+
+        unroll_binary_op(lhs, rhs, sub.clone(), sub_ref.span())
+    }
+    fn visit_mul_bis(
+        &mut self,
+        _graph: &mut Graph,
+        mul: Link<Op>,
+    ) -> Result<Option<Link<Op>>, CompileError> {
+        // safe to unwrap because we just dispatched on it
+        let mul_ref = mul.as_mul().unwrap();
+        let lhs = mul_ref.lhs.clone();
+        let rhs = mul_ref.rhs.clone();
+
+        unroll_binary_op(lhs, rhs, mul.clone(), mul_ref.span())
+    }
+    fn visit_exp_bis(
+        &mut self,
+        _graph: &mut Graph,
+        exp: Link<Op>,
+    ) -> Result<Option<Link<Op>>, CompileError> {
+        // safe to unwrap because we just dispatched on it
+        let exp_ref = exp.as_exp().unwrap();
+        let lhs = exp_ref.lhs.clone();
+        let rhs = exp_ref.rhs.clone();
+
+        unroll_binary_op(lhs, rhs, exp.clone(), exp_ref.span())
+    }
+
+    fn visit_enf_bis(
+        &mut self,
+        _graph: &mut Graph,
+        enf: Link<Op>,
+    ) -> Result<Option<Link<Op>>, CompileError> {
+        let mut updated_enf = None;
+
+        {
+            let enf_ref = enf.as_enf().unwrap();
+            let expr = enf_ref.expr.clone();
+            if let Op::Vector(vec) = expr.borrow().deref() {
+                let ops = vec.children().borrow().deref().clone();
+                let mut new_vec = vec![];
+                for op in ops.iter() {
+                    let new_node = Enf::create(op.clone(), enf_ref.span());
+                    new_vec.push(new_node);
+                }
+                updated_enf = Some(Vector::create(new_vec, enf_ref.span()));
+            };
+        }
+
+        Ok(updated_enf)
+    }
+
+    fn visit_boundary_bis(
+        &mut self,
+        _graph: &mut Graph,
+        boundary: Link<Op>,
+    ) -> Result<Option<Link<Op>>, CompileError> {
+        let mut updated_boundary = None;
```

Comment:
Nit(also applicable to the other visit functions): this variable can be omitted by returning the result in the `if` branch.